test/extended/util/prometheus/helpers.go

```diff
@@ -193,10 +193,26 @@ func RunQuery(ctx context.Context, prometheusClient prometheusv1.API, query stri
 }
 
 func RunQueryAtTime(ctx context.Context, prometheusClient prometheusv1.API, query string, evaluationTime time.Time) (*PrometheusResponse, error) {
-	result, warnings, err := prometheusClient.Query(ctx, query, evaluationTime)
-	if err != nil {
-		return nil, err
+	var lastErr error
+	var result model.Value
+	var warnings prometheusv1.Warnings
+	for i := 0; i < 5; i++ {
+		if ctx.Err() != nil {
+			return nil, ctx.Err()
+		}
+		result, warnings, lastErr = prometheusClient.Query(ctx, query, evaluationTime)
+		if lastErr != nil {
+			select {
+			case <-time.After(10 * time.Second):
+			case <-ctx.Done():
+				break
+			}
+		}
```

Comment:
shouldn't it break from the for loop?
```suggestion
		}
		break
```

Reply:
sorry for being unclear, I meant to say that we don't seem to break from the main for loop in case no error is returned.

Reply:
yeah, how embarrassing. I actually dreamed this one last night.